server/migrations/20231031162340-verifyNoSkippedTests.js

```diff
@@ -0,0 +1,67 @@
+'use strict';
+
+/** @type {import('sequelize-cli').Migration} */
+module.exports = {
+    async up(queryInterface /* , Sequelize */) {
+        return queryInterface.sequelize.transaction(async transaction => {
+            const [rows] = await queryInterface.sequelize.query(
+                `
+                    SELECT
+                        "TestPlanReport".id,
+                        "TestPlanReport"."atId",
+                        "TestPlanRun"."testResults",
+                        "TestPlanVersion".tests
+                    FROM
+                        "TestPlanReport"
+                        INNER JOIN "TestPlanVersion" ON "TestPlanVersion".id = "TestPlanReport"."testPlanVersionId"
+                        INNER JOIN "TestPlanRun" ON "TestPlanReport".id = "TestPlanRun"."testPlanReportId"
+                    WHERE
+                        "TestPlanReport"."markedFinalAt" IS NOT NULL
+                `,
+                { transaction }
+            );
+
+            const dataById = {};
+            rows.forEach(({ id, atId, testResults, tests }) => {
+                if (dataById[id]) {
+                    dataById[id].runLengths.push(testResults.length);
+                } else {
+                    const runnableTestsLength = tests.filter(test =>
+                        test.atIds.includes(atId)
+                    ).length;
+                    dataById[id] = {
+                        id,
+                        runnableTestsLength,
+                        runLengths: [testResults.length]
+                    };
+                }
+            });
+
+            for (const data of Object.values(dataById)) {
+                let isComplete = true;
+
+                data.runLengths.forEach(runLength => {
+                    if (runLength !== data.runnableTestsLength) {
+                        isComplete = false;
+                    }
+                });
```

Comment:
```suggestion
                    const isComplete = data.runLengths.every(runLength => runLength === data.runnableTestsLength);

```
Another instance where it could be useful to use `.every` for an affirmative calculation in one line.